lib/src/chunked_stream_reader.dart

```diff
@@ -97,29 +98,38 @@ class ChunkedStreamReader<T> {
       // While we have data to read
       while (size > 0) {
         // Read something into the buffer, if it's empty
-        if (_buffer.isEmpty) {
+        if (_buffer.length - _offset <= 0) {
           if (!(await _input.moveNext())) {
             // Don't attempt to read more data, as there is no more data.
             size = 0;
             _reading = false;
             break;
           }
           _buffer = _input.current;
+          _offset = 0;
         }
 
-        if (_buffer.isNotEmpty) {
-          if (size < _buffer.length) {
-            final output = _buffer.sublist(0, size);
-            _buffer = _buffer.sublist(size);
+        if (_buffer.length - _offset > 0) {
+          if (size < _buffer.length - _offset) {
+            late List<T> output;
```

Comment:
It looks like this doesn't actually need to be `late`, flow analysis figures it out.

Reply:
Oh, null-safety is really cool... maybe we need a hint / lint for cases like this -- I didn't know, and I've seen this elsewhere.